lib/proxy_api/proxy_exception.rb

```diff
@@ -3,12 +3,20 @@ class ProxyException < ::Foreman::WrappedException
     attr_reader :url
 
     def initialize(url, exception, message, *params)
+      message += append_exception_message(exception)
```

Comment:
Why append the message in this location and not in the `message` method? I think this will affect the error codes generated, which will now depend on the passed message and _any_ text supplied in the smart proxy response.